api/internal/managers/app/config/config.go

```diff
@@ -196,3 +235,23 @@ func filterAppConfig(config kotsv1beta1.Config) (kotsv1beta1.Config, error) {
 func isItemEnabled(when multitype.QuotedBool) bool {
 	return when != "false"
 }
+
+// isRequiredItem checks if an item is required based on whether Required is true and the item is
+// enabled and not hidden
+func isRequiredItem(item kotsv1beta1.ConfigItem) bool {
+	if !item.Required {
+		return false
+	}
+	if !isItemEnabled(item.When) {
+		return false
+	}
+	if item.Hidden {
+		return false
+	}
```

Comment:
I created a follow up story for this https://app.shortcut.com/replicated/story/126644/correct-app-config-hidden-required-behavior